### ISSUE TITLE
Rendering style tag with stache template

### ISSUE THREAD

**Issue by whitecolor** [canjs/canjs#2035]
`style` tag internal content seem to be blocked from rendering stache template.

```javascript
var map = new can.Map({color: 'green'})
$('head').append(can.stache('<style>body: {{color}}</style>')(map))
```

the result in HTML: `<style>body: {{color}}</style>'`

This works:
```javascript
var map = new can.Map({color: 'green'})
$('head').append(can.stache('<xxx>body: {{color}}</xxx>')(map))
```

the result in HTML: `<xxx>body: green</xxx>'` - and it is dynamic (bound to `map.color` changes)

**Comment by marshallswain** [comment]
I've noticed this before.  I used a helper with triple curly braces {{{layoutStyles}}} to generate a dynamic style.

**Comment by whitecolor** [comment]
@marshallswain well that works
```
var map = new can.Map({color: 'green'})
$('body').append(can.stache('{{{colorHelper}}}')(map, {colorHelper: function(){return '<style>body{color: ' + map.attr('color') +'}</style>'}}))
```
but it is not very convinient

**Comment by justinbmeyer** [comment]
Style needs to be special cased similar to how textareas are special cased.

**Comment by whitecolor** [comment]
@justinbmeyer @marshallswain 

What do you think about making styleshits dynamic using stache templtes? it could allow intresting technics of styling components with dymics css properties (that would come from scope) or even mixins (made using helpers).

**Comment by justinbmeyer** [comment]
@whitecolor I think it's an interesting idea.  I wonder what advantageous it would have over simply changing an element's `style` properties?

To do this right, you'd want a CSS parser, and a CSS target, and cssRule can.view.live method.  This way, the updates to [CSS rules](https://developer.mozilla.org/en-US/docs/Web/API/CSSRule) can happen efficiently.  

Browser support with CSS rules is a bit all over the place I think if you go back far enough.

**Comment by marshallswain** [comment]
@whitecolor I think it's a really good idea.  I'm not sure how often I would use it, other than with the occasional tricky layout.  I agree that using the helper syntax isn't very fun, but it doesn't look quite as ridiculous when you use it in a component.  :)

**Comment by justinbmeyer** [comment]
@whitecolor you mean:

```
$('head').append(can.stache('<style>body { {{color}} }</style>')(map))
```
right?

**Comment by justinbmeyer** [comment]
Fixed this in #2083.  @whitecolor check it out.
